src/test/java/com/microsoft/sqlserver/jdbc/bulkCopy/BulkCopyTestUtil.java

```diff
@@ -280,17 +281,26 @@ static void validateValues(DBConnection con, DBTable sourceTable, DBTable destin
             int totalColumns = destMeta.getColumnCount();
 
             // verify data from sourceType and resultSet
+            int numRows = 0;
             while (srcResultSet.next() && dstResultSet.next()) {
+                numRows++;
                 for (int i = 1; i <= totalColumns; i++) {
-                    // TODO: check row and column count in both the tables
-
                     Object srcValue, dstValue;
                     srcValue = srcResultSet.getObject(i);
                     dstValue = dstResultSet.getObject(i);
 
                     ComparisonUtil.compareExpectedAndActual(destMeta.getColumnType(i), srcValue, dstValue);
                 }
             }
+
+            // verify number of rows and columns
+            assertTrue(((ResultSet) srcResultSet.product()).getMetaData().getColumnCount() == totalColumns);
+            assertTrue(((ResultSet) dstResultSet.product()).getMetaData().getColumnCount() == totalColumns);
+
+            if (numRows > 0) {
```

Comment:
Shouldn't the test fail if `numRows` is 0?

Reply:
no, because this method is used sometimes to just copy a table with no data